src/xhydro/modelling/_ravenpy_models.py

```diff
@@ -78,6 +80,7 @@ def __init__(
         data_type,
         alt_names_meteo,
         meteo_station_properties,
+        hydro_station_properties={},
```

Comment:
This signature creates a mutable default argument, meaning that multiple instances of the `RavenPyModel` running in the same script could inherit changes to this argument on prior initialisation. See: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Another way to do this would be 
```suggestion
        hydro_station_properties: dict | None = None,
```
Followed by:
```python
if hydro_station_properties is None:
    hydro_station_properties = {}
```